File: java/example_code/cloudwatch/src/main/java/aws/example/cloudwatch/DeleteAlarm.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
package aws.example.cloudwatch;

import com.amazonaws.services.cloudwatch.AmazonCloudWatch;
import com.amazonaws.services.cloudwatch.AmazonCloudWatchClientBuilder;
import com.amazonaws.services.cloudwatch.model.DeleteAlarmsRequest;
import com.amazonaws.services.cloudwatch.model.DeleteAlarmsResult;

/**
 * Deletes a CloudWatch alarm
 */
public class DeleteAlarm {
    public static void main(String[] args) {

        final String USAGE = "To run this example, supply an alarm name\n" +
                "Ex: DeleteAlarm <alarm-name>\n";

        if (args.length != 1) {
            System.out.println(USAGE);
            System.exit(1);
        }

        String alarm_name = args[0];

        final AmazonCloudWatch cw = AmazonCloudWatchClientBuilder.defaultClient();

        DeleteAlarmsRequest request = new DeleteAlarmsRequest()
                .withAlarmNames(alarm_name);

        DeleteAlarmsResult response = cw.deleteAlarms(request);

        System.out.printf("Successfully deleted alarm %s", alarm_name);
    }
}

File: .kotlin_alpha/services/cloudwatch/src/main/kotlin/com/kotlin/cloudwatch/DeleteAlarm.kt
//snippet-sourcedescription:[DeleteAlarm.kt demonstrates how to delete an Amazon CloudWatch alarm.]
//snippet-keyword:[AWS SDK for Kotlin]
//snippet-keyword:[Code Sample]
//snippet-service:[Amazon CloudWatch]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[06/11/2021]
//snippet-sourceauthor:[scmacdon - aws]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.cloudwatch

// snippet-start:[cloudwatch.kotlin.delete_metrics.import]
import aws.sdk.kotlin.services.cloudwatch.CloudWatchClient
import aws.sdk.kotlin.services.cloudwatch.model.DeleteAlarmsRequest
import aws.sdk.kotlin.services.cloudwatch.model.CloudWatchException
import kotlin.system.exitProcess
// snippet-end:[cloudwatch.kotlin.delete_metrics.import]

suspend fun main(args:Array<String>) {

    val usage  = """

    Usage:
        <alarmName>  

    Where:
        alarmName - an alarm name to delete.
     """

    if (args.size != 1) {
        println(usage)
        exitProcess(0)
    }

    val alarmName = args[0]
    val cwClient = CloudWatchClient{region="us-east-1"}
    deleteCWAlarm(cwClient, alarmName)
    cwClient.close()
}

// snippet-start:[cloudwatch.kotlin.delete_metrics.main]
suspend fun deleteCWAlarm(cwClient: CloudWatchClient, alarmNameVal: String) {

    try {
        val request = DeleteAlarmsRequest {
            alarmNames = listOf(alarmNameVal)
        }

        cwClient.deleteAlarms(request)
        println("Successfully deleted alarm $alarmNameVal")

    } catch (ex: CloudWatchException) {
        println(ex.message)
        cwClient.close()
        exitProcess(0)
    }
}
// snippet-end:[cloudwatch.kotlin.delete_metrics.main]